test/test_ops.py

```diff
@@ -1102,296 +1101,374 @@ def test_bbox_convert_jit(self):
-    box1 = torch.tensor([-1, -1, 1, 1], dtype=dtype, device=device)
-    box2 = torch.tensor([0, 0, 1, 1], dtype=dtype, device=device)
-    box3 = torch.tensor([0, 1, 1, 2], dtype=dtype, device=device)
-    box4 = torch.tensor([1, 1, 2, 2], dtype=dtype, device=device)
 
-    box1s = torch.stack(
-        [box2, box2],
-        dim=0,
-    )
-    box2s = torch.stack(
-        [box3, box4],
-        dim=0,
-    )
+class TestCompleteBoxIouLoss:
+    @pytest.mark.parametrize("dtype", [torch.float32, torch.half])
+    @pytest.mark.parametrize("device", cpu_and_gpu())
+    def test_ciou_loss(self, dtype, device):
+        box1, box2, box3, box4, box1s, box2s = get_boxes(dtype, device)
 
-    def assert_distance_iou_loss(box1, box2, expected_output, reduction="none"):
-        output = ops.distance_box_iou_loss(box1, box2, reduction=reduction)
-        # TODO: When passing the dtype, the torch.half fails as usual.
-        expected_output = torch.tensor(expected_output, device=device)
-        tol = 1e-5 if dtype != torch.half else 1e-3
-        torch.testing.assert_close(output, expected_output, rtol=tol, atol=tol)
+        assert_iou_loss(ops.complete_box_iou_loss, box1, box1, 0.0, device=device)
+        assert_iou_loss(ops.complete_box_iou_loss, box1, box2, 0.8125, device=device)
+        assert_iou_loss(ops.complete_box_iou_loss, box1, box3, 1.1923, device=device)
+        assert_iou_loss(ops.complete_box_iou_loss, box1, box4, 1.2500, device=device)
+        assert_iou_loss(ops.complete_box_iou_loss, box1s, box2s, 1.2250, device=device, reduction="mean")
+        assert_iou_loss(ops.complete_box_iou_loss, box1s, box2s, 2.4500, device=device, reduction="sum")
 
-    assert_distance_iou_loss(box1, box1, 0.0)
+    @pytest.mark.parametrize("device", cpu_and_gpu())
+    @pytest.mark.parametrize("dtype", [torch.float32, torch.half])
+    def test_empty_inputs(self, dtype, device):
+        assert_empty_loss(ops.complete_box_iou_loss, dtype, device)
 
-    assert_distance_iou_loss(box1, box2, 0.8125)
 
-    assert_distance_iou_loss(box1, box3, 1.1923)
+class TestDistanceBoxIouLoss:
+    @pytest.mark.parametrize("device", cpu_and_gpu())
+    @pytest.mark.parametrize("dtype", [torch.float32, torch.half])
+    def test_distance_iou_loss(self, dtype, device):
+        box1, box2, box3, box4, box1s, box2s = get_boxes(dtype, device)
 
-    assert_distance_iou_loss(box1, box4, 1.2500)
+        assert_iou_loss(ops.distance_box_iou_loss, box1, box1, 0.0, device=device)
+        assert_iou_loss(ops.distance_box_iou_loss, box1, box2, 0.8125, device=device)
+        assert_iou_loss(ops.distance_box_iou_loss, box1, box3, 1.1923, device=device)
+        assert_iou_loss(ops.distance_box_iou_loss, box1, box4, 1.2500, device=device)
+        assert_iou_loss(ops.distance_box_iou_loss, box1s, box2s, 1.2250, device=device, reduction="mean")
+        assert_iou_loss(ops.distance_box_iou_loss, box1s, box2s, 2.4500, device=device, reduction="sum")
```

Comment:
Future improvement: I know that these values were also used originally but I think it's problematic that both cIoU and dIoU losses are being tested against examples that yield identical outputs. Aka if I send a PR that replaces cIoU with dIoU, how do we capture the issue?